configs/projects/puppet-agent.rb

```diff
@@ -241,7 +241,20 @@
   end
   proj.component "ruby-shadow" unless platform.is_aix? || platform.is_windows?
   proj.component "ruby-augeas" unless platform.is_windows?
-  proj.component "openssl"
+
+  if ENV["USE_SYSTEM_OPENSSL"] == '1'
+    if (platform.is_el? || platform.is_fedora?)
+      # Currently we plan to support linking against system openssl
+      # only on these platforms.
+      proj.setting(:vendor_openssl, false)
+    else
+      fail "Currently linking against system openssl is supported only on RHEL family platforms"
```

Comment:
Nitpick: I'd explicitly state "Enterprise Linux or Fedora platforms". Abbreviations are OK in the code paths but the UX should be as explicitly precise as possible.